platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolutionstrategy/DefaultCapabilitiesResolution.java

```diff
@@ -120,6 +121,14 @@ private void selectExplicitCandidate(DefaultCapabilityResolutionDetails resoluti
                 String reason = resolutionDetails.reason;
                 if (reason != null) {
                     cand.byReason(Describables.of("On capability", version.getCapabilityId(), reason));
+                } else {
+                    cand.byReason(() -> {
+                        String selectedNodeDesc = resolutionDetails.selected.getId().getDisplayName();
+                        if (resolutionDetails.selected.getId() instanceof ProjectComponentIdentifier) {
+                            selectedNodeDesc += " variant " + resolutionDetails.selected.getVariantName();
+                        }
```

Comment:
❓ Can non-project dependencies also have a meaningful variant name?

Maybe we could add a `getDisplayName()` on `ComponentVariantIdentifier`?

Reply:
We can ask for the variant name here and it would be meaningful. 

However, it can be annoying to test, as the variant name changes between ivy and maven.

Reply:
I bit the bullet and dealt with the annoying test expectations. 

We now print the variant name here always